scripts/portal-flow.js

```diff
@@ -1,7 +1,87 @@
-module.exports = function(runner) {
+'use strict';
+const configureRequest = require('../util/configureRequest');
+const requestBodyUtils = require('../util/sync_request_bodies');
+const recordUtils = require('../util/generate_record');
+const makeUser = require('../util/fixtures/makeUser');
+const makeWorkorder = require('../util/fixtures/makeWorkorder');
+const makeWorkflow = require('../util/fixtures/makeWorkflow');
+const makeMessage = require('../util/fixtures/makeMessage');
+const createUserAndGroup = require('../util/createUserAndGroup');
+const Promise = require('bluebird');
+
+function urlFor(baseUrl, dataset) {
+  return `${baseUrl}/mbaas/sync/${dataset}`;
+}
+
+function syncDataset(baseUrl, request, clientId, name) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    dataset_id: name,
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: []
+  });
+  return request.post({
+    url: urlFor(baseUrl, name),
+    body: payload,
+    json: true
+  }).then();
+}
+
+function createRecord(baseUrl, request, clientId, dataset, data) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    fn: 'sync',
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: [recordUtils.generateRecord(data)]
+  });
+  return request.post({
+    url: urlFor(baseUrl, 'workorders'),
```

Comment:
The hardcoded `'workorders'` here needs to change to `dataset`, otherwise we just get lots of garbage workorders. I noticed this when trying out this code.

Should the raincatcher modules have some sort of guard against non-conforming data? E.g. "wat is this, it doesn't look like a workorder, I'll ignore it"